src/lib/utils/mem_ops.h

```diff
@@ -61,24 +63,57 @@ class Allocator_Initializer final {
 */
 BOTAN_PUBLIC_API(2, 0) void secure_scrub_memory(void* ptr, size_t n);
 
+/**
+* Scrub memory contents in a way that a compiler should not elide,
+* using some system specific technique. Note that this function might
+* not zero the memory.
+*
+* @param data  the data region to be scrubbed
+*/
+void secure_scrub_memory(ranges::contiguous_output_range auto&& data) {
+   secure_scrub_memory(std::ranges::data(data), ranges::size_bytes(data));
+}
```

Comment:
TIL: simple concept annotations can be added to parameters without the verbose `template<typename T> requires some_concept<T>` syntax.

Reply:
Nice!